clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -54,10 +54,12 @@ namespace direct {
 namespace {
 /// If the given type is a vector type, return the vector's element type.
 /// Otherwise return the given type unchanged.
-// TODO(cir): Return the vector element type once we have support for vectors
-// instead of the identity type.
 mlir::Type elementTypeIfVector(mlir::Type type) {
-  assert(!cir::MissingFeatures::vectorType());
+  if (auto vecType = mlir::dyn_cast<cir::VectorType>(type))
+    return vecType.getElementType();
+
+  if (auto vecType = mlir::dyn_cast<mlir::VectorType>(type))
```

Comment:
I've looked at this twice now, and both times I thought these two `if` statements were identical. It took a third look to see that one is checking for `cir::VectorType` and the other is checking for `mlir::VectorType`. Maybe add a comment drawing attention to this fact?

Reply:
Yes, it was confusing for me too, especially when I got the exception, it was weird why it didn't return the element type :D

Reply:
Can you implement this with a TypeSwitch?

```
return llvm::TypeSwitch<mlir::Type, mlir::Type>(type)
           .Case<cir::VectorType, mlir::VectorType>([](auto p) { return p->getElementType(); }
           .Default([](mlir::Type p) { return p; });
```

Reply:
Done